dask/array/reductions.py

```diff
@@ -38,7 +38,8 @@ def divide(a, b, dtype=None):
 
 
 def reduction(x, chunk, aggregate, axis=None, keepdims=None, dtype=None,
-              split_every=None, combine=None, name=None, out=None):
+              split_every=None, combine=None, name=None, out=None,
+              concatenate=True, output_size=1):
```

Comment:
Adding two new keyword arguments to this function seems important.  Can I ask you to explain why they were necessary, perhaps in the top comment of this PR? 

Also, I *wanted* to ask you to also add them to the docstring, but I see that the current docstring is unfortunately quite sprase.  Git blame shows this to be my fault :/

Reply:
I'm writing the whole docstring...

I'd also like to change the function signature to ``reduction(x, chunk, combine, concatenate, ...)`` as I feel the concatenate function is fundamental and should not be relegated to a kwarg. Also it makes it easier for the user to wrap his head around it if the order of the arguments matchhes the order of execution.
Do you agree to the change? I would make it in a separate PR to keep things clean.

Reply:
edited typos